metrics/metrics.go

```diff
@@ -45,32 +46,42 @@ func Middleware(registry prometheus.Registerer) gin.HandlerFunc {
 		Name:      "request_duration_seconds",
 		Help:      "A histogram of duration, in seconds, handling HTTP requests.",
 		Buckets:   prometheus.ExponentialBuckets(0.001, 2, 15),
-	}, []string{"host", "method", "path", "status"})
+	}, []string{"host", "method", "path", "status", "blocking"})
 
-	requestCount := prometheus.NewGauge(prometheus.GaugeOpts{
+	requestCount := prometheus.NewGaugeVec(prometheus.GaugeOpts{
 		Namespace: "http",
 		Name:      "requests_active",
 		Help:      "A gauge of the number of requests currently being handled.",
-	})
+	}, []string{"blocking"})
 
 	registry.MustRegister(requestDuration, requestCount)
 
 	return func(c *gin.Context) {
-		requestCount.Inc()
-		defer requestCount.Dec()
+		blocking := blockingRequestLabel(c.Request)
+
+		count := requestCount.With(prometheus.Labels{"blocking": blocking})
+		count.Inc()
+		defer count.Dec()
 
 		t := time.Now()
 		c.Next()
 
 		requestDuration.With(prometheus.Labels{
```

Comment:
Does it even make sense to have request duration metrics for blocking requests? Perhaps this can be omitted if `blocking=true`?